### PR TITLE
try runtime

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
may be this https://github.com/AcalaNetwork/chopsticks/issues/70 (so only None , no checks filter)

### CODE REVIEW

File: packages/chopsticks/src/try-runtime.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import { generateHtmlDiffPreviewFile } from './utils/generate-html-diff'
+import { openHtml } from './utils/open-html'
+import { runTask, taskHandler } from './executor'
+import { setup } from './setup'
+
+export const tryRuntime = async (argv: Config) => {
+  const context = await setup(argv)
+  const wasm = await context.chain.head.wasm
+  const block = context.chain.head
+  const parent = await block.parentBlock
```
Comment (ermalkaleci) [inline packages/chopsticks/src/try-runtime.ts:14]: why do you need parent?

File: packages/chopsticks/src/cli.ts [hunk 2]
```diff
@@ -81,6 +82,33 @@ yargs(hideBin(process.argv))
       await runBlock(await processArgv(argv))
     }
   )
+yargs(hideBin(process.argv))
```
Comment (ermalkaleci) [inline packages/chopsticks/src/cli.ts:85]: you don't need those 2 lines
```suggestion
```

File: packages/chopsticks/src/cli.ts [hunk 3]
```diff
@@ -81,6 +82,33 @@ yargs(hideBin(process.argv))
...
+  .scriptName('chopsticks')
+  .command(
+    'try-runtime',
+    'Runs runtime upgrade',
+    (yargs) =>
+      yargs.options({
+        ...defaultOptions,
```
Comment (ermalkaleci) [inline packages/chopsticks/src/cli.ts:92]: we need to make wasm-override required for this command

File: packages/chopsticks/src/cli.ts [hunk 4]
```diff
@@ -81,6 +82,33 @@ yargs(hideBin(process.argv))
...
+        port: {
```
Comment (ermalkaleci) [inline packages/chopsticks/src/cli.ts:93]: no port is required

File: packages/chopsticks/src/try-runtime.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    UpgradeCheckSelect: {
+      _enum: {
+        None: null,
+      },
+    },
+  })
+
+  const select_none = registry.createType('UpgradeCheckSelect', 'None')
+  const calls: [string, HexString[]][] = [['TryRuntime_on_runtime_upgrade', [select_none.toHex()]]]
+  const result = await runTask(
```
Comment (ermalkaleci) [inline packages/chopsticks/src/try-runtime.ts:27]:
```suggestion
  const result = await block.call('TryRuntime_on_runtime_upgrade', [select_none.toHex()])
```

File: packages/chopsticks/src/cli.ts (hunk 3 above)
Comment (dzmitry-lahoda) [reply packages/chopsticks/src/cli.ts:92]: i am not sure about it . in theory i can add data to chain and try runtime. without chaning wasm.

Comment (ermalkaleci) [reply packages/chopsticks/src/cli.ts:90]: But you need to provide a wasm build with feature "try-runtime" right?

File: packages/chopsticks/src/cli.ts [hunk 6]
```diff
@@ -81,6 +82,27 @@ yargs(hideBin(process.argv))
       await runBlock(await processArgv(argv))
     }
   )
+  .command(
+    'try-runtime',
+    'Runs runtime upgrade',
+    (yargs) =>
+      yargs.options({
+        ...defaultOptions,
+        'output-path': {
```
Comment (ermalkaleci) [inline packages/chopsticks/src/cli.ts:91]: add it here please
```suggestion
'wasm-override': {
    desc: '...',
    string: true,
    required: true
  },
        'output-path': {
```

Comment (dzmitry-lahoda) [reply packages/chopsticks/src/cli.ts:91]: so it will not check override in config? so we have 2 places of checks. is 2 checks okey?

Comment (ermalkaleci) [reply packages/chopsticks/src/cli.ts:91]: I mean remove check from tryRuntime function and add it here. `wasm-override is` defined on `...defaultOptions` but it's optional. we are making it required for try-runtime command

Comment (ermalkaleci) [reply packages/chopsticks/src/cli.ts:91]: Having it on cli command will give you the benefit of help `yarn start try-runtime --help`

Comment (ermalkaleci) [reply packages/chopsticks/src/cli.ts:91]: @dzmitry-lahoda can you do this last change so we can merge?

**Pull request merged**